x-pack/plugins/triggers_actions_ui/README.md

```diff
@@ -0,0 +1,1054 @@
+# Kibana Alerts and Actions UI
+
+The Kibana alerts and actions UI plugin provides a user interface for managing alerts and actions. 
+As a developer you can reuse and extend buildin alerts and actions UI functionality:
+
+- Create and register new alert type.
+- Create and register new action type.
+- Embed Create Alert flyout to Kibana plugins.
+
+To enable Alerts and Actions UI functionality, the next configuration settings should be provided:
+```
+xpack.alerting.enabled: true
+xpack.actions.enabled: true
+xpack.triggers_actions_ui.enabled: true
+xpack.triggers_actions_ui.createAlertUiEnabled: true
+```
+
+-----
+
+
+Table of Contents
+
+- [Kibana Alerts and Actions UI](#kibana-alerts-and-actions-ui)
+  - [Build and register Alert Types](#build-and-register-alert-types)
+    - [Built-in Alert Types](#built-in-alert-types)
+      - [Index Threshold Alert](#index-threshold-alert)
+    - [Alert type model definition](#alert-type-model-definition)
+    - [Register alert type model](#register-alert-type-model)
+    - [Create and register new alert type UI example](#create-and-register-new-alert-type-ui-example)
+    - [Common expression components](#common-expression-components)
+      - [WHEN expression component](#when-expression-component)
+      - [OF expression component](#of-expression-component)
+      - [GROUPED BY expression component](#grouped-by-expression-component)
+      - [FOR THE LAST expression component](#for-the-last-expression-component)
+      - [THRESHOLD expression component](#threshold-expression-component)
```

Comment:
I'm wondering if maybe we should extract the Expression types into a separate guide?
They feel more generic that the Alerting UI, no?